server/game/core/gameSteps/ActionWindow.js

```diff
@@ -190,6 +190,20 @@ class ActionWindow extends UiPrompt {
         super.complete();
     }
 
+    highlightSelectableCards() {
+        const allPossibleCards = this.game.findAnyCardsInPlay().concat(
+            this.activePlayer.getCardPile(Location.Discard),
+            this.activePlayer.getCardPile(Location.Resource),
+            this.activePlayer.getCardPile(Location.Hand)
+        );
+        this.activePlayer.setSelectableCards(allPossibleCards.filter((card) => this.getCardLegalActions(card, this.activePlayer).length > 0));
+    }
+
+    getCardLegalActions(card, player) {
+        let actions = card.getActions();
```

Comment:
Since this is a really common debug point let's split this out a little bit to make the values easier to see:

```typescript
getCardLegalActions(card, player) {
    const actions = card.getActions();
    const legalActions = actions.filter((action) => action.meetsRequirements(action.createContext(player)) === '')
    return legalActions;
}
```

That way folks can just break at the return statement and see `actions` and `legalActions` in the debugger

Reply:
hah, that's actually what I've been doing to see what's happening here. I would love to break it up.